src/IconButton/index.jsx

```diff
@@ -79,7 +86,58 @@ IconButton.propTypes = {
   onClick: PropTypes.func,
   /** Type of button (uses Bootstrap options) */
   variant: PropTypes.oneOf(['primary', 'secondary', 'success', 'warning', 'danger', 'light', 'dark', 'black']),
+  /** size of button to render */
   size: PropTypes.oneOf(['sm', 'md', 'inline']),
+  /** whether to show the IconButton in an active state, whose styling is distinct from default state */
+  isActive: PropTypes.bool,
 };
 
+/**
+ *
+ * @param { object } args Arguments
+ * @param { string } args.tooltipPlacement choose from https://popper.js.org/docs/v2/constructors/#options
+ * @param { React.Component } args.tooltipContent any content to pass to tooltip content area
+ * @returns { IconButton } a button wrapped in overlaytrigger
+ */
+const WithTooltip = ({
+  tooltipPlacement, tooltipContent, variant, invertColors, ...props
+}) => {
+  const invert = invertColors ? 'inverse-' : '';
+  return (
+    <OverlayTrigger
+      key={tooltipPlacement}
+      placement={tooltipPlacement}
+      overlay={(
+        <Tooltip
+          id={`iconbutton-tooltip-${tooltipPlacement}`}
+          variant={invert ? 'light' : ''}
+        >
+          {tooltipContent}
+        </Tooltip>
+      )}
+    >
+      <IconButton variant={variant} invertColors={invertColors} {...props} />
+    </OverlayTrigger>
+  );
+};
+
+WithTooltip.defaultProps = {
+  tooltipPlacement: 'top',
+  variant: 'primary',
+  invertColors: false,
+};
+
+WithTooltip.propTypes = {
+  /** tooltip placement can be top, left, right etc, per https://popper.js.org/docs/v2/constructors/#options  */
+  tooltipPlacement: PropTypes.string,
+  /** any valid JSX or text to be rendered as tooltip contents */
+  tooltipContent: PropTypes.node.isRequired,
+  /** Type of button (uses Bootstrap options) */
+  variant: PropTypes.oneOf(['primary', 'secondary', 'success', 'warning', 'danger', 'light', 'dark', 'black']),
+  /** Changes icon styles for dark background */
+  invertColors: PropTypes.bool,
+};
+
+IconButton.WithTooltip = WithTooltip;
+
```

Comment:
We should also export `WithTooltip` as a named export here and in `src/index.js` so that consumers can also import it directly, e.g., `import { WithTooltip } from '@edx/paragon';`.

However, this name `WithTooltip` doesn't really accurately describe the component on its own. I might recommend renaming it to `IconButtonWithTooltip`. What do you think?

Reply:
ah ok missed this, yes if we export as independent component, we need to rename it yeah

Reply:
exporting as `export { IconButtonWithTooltip };` now into index.js